libsnark/zk_proof_systems/plonk/srs.tcc

```diff
@@ -70,19 +70,68 @@ plonk_keypair<ppT>::plonk_keypair(
 {
 }
 
-/// struct transcript_hash_t constructor
+/// transcript_hasher constructor
 template<typename ppT>
-transcript_hash_t<ppT>::transcript_hash_t(
-    libff::Fr<ppT> &beta,
-    libff::Fr<ppT> &gamma,
-    libff::Fr<ppT> &alpha,
-    libff::Fr<ppT> &zeta,
-    libff::Fr<ppT> &nu,
-    libff::Fr<ppT> &u)
-    : beta(beta), gamma(gamma), alpha(alpha), zeta(zeta), nu(nu), u(u)
+transcript_hasher<ppT>::transcript_hasher(size_t &istep) : istep(istep)
 {
```

Comment:
Would we set this to anything other than 0?  Why doesn't the constructor just create an array of the test vector values, ad we read the `step`-th one?  This will also simplify `get_hash` which seems to have a lot of repeated code.

Reply:
Created an array with test values initialized by the constructor as suggested in the comment.

Regarding the `istep` variable, it has been removed since now we are using the length of the buffer as the indicator which challenge to output.